bip78/src/receiver/mod.rs

```diff
@@ -14,6 +13,18 @@ pub struct UncheckedProposal {
     psbt: Psbt,
 }
 
+pub struct MaybeInputsOwned {
+    psbt: Psbt,
```

Comment:
We will also have to carry around information from HTTP request.

Reply:
certainly, receiver parse & application of sender proposal params are in scope for the next pr

Reply:
https://github.com/Kixunil/payjoin/pull/30

Reply:
Ah, I now think I understood what you intended in the other PR. You wanted to reuse the `Params` struct here.

I suggest:

* Keep `Params` `pub(crate)` and move to a common module
* What currently is exposed as `Params` should be named `Configuration`, it can hold `Params` inside
* Consumers can manipulate `Configuration` via builder API but not `Params` directly.

Reply:
This was the comment I looking for. I feel like I'm getting the ultimate trial by fire rust experience here :D

Reply:
Yeah, I can see many details. :) Hope it helps you get more skilled.